course-definition.yml

```
@@ -81,6 +80,10 @@ stages:
 
       You program must [exit](https://en.wikipedia.org/wiki/Exit_status) with 0 if the character is found, and 1 if not.
 
+      ### Notes
+
+      - To learn how grep works under the hood, you'll build your own regex matcher from scratch instead of using {{language_name}}'s built-in regex features.
```

```suggestion
      - To learn how Regexes work under the hood, you'll build your own regex implementation from scratch instead of using {{language_name}}'s built-in regex features.
```

Let's keep focus on "how regexes work", not "how grep works" - and let's say "regex implementation" instead of "regex matcher" to be a bit more concrete